tests/shared/s3-packages.spec.js

```diff
@@ -0,0 +1,1012 @@
+          done();
+        }).catch(done);
+      });
+
+    });
+
+    describe('given a greater snapshot version in a production bucket', function() {
+
+      beforeEach(function() {
+        this.getRemoteVersionsStub = m.sinon.stub(s3Packages, 'getRemoteVersions');
+        this.getRemoteVersionsStub.returns(Bluebird.resolve([
+          '1.0.0',
+          '2.0.0+abb6139'
+        ]));
+      });
+
+      afterEach(function() {
+        this.getRemoteVersionsStub.restore();
+      });
+
+      it('should ignore snapshot versions', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('1.0.0');
+          done();
+        }).catch(done);
+      });
+
+    });
+
+    describe('given production v1, v2, and v3 remote versions', function() {
+
+      beforeEach(function() {
+        this.getRemoteVersionsStub = m.sinon.stub(s3Packages, 'getRemoteVersions');
+        this.getRemoteVersionsStub.returns(Bluebird.resolve([
+          '3.0.1',
+          '3.0.0',
+          '2.1.1',
+          '2.1.0',
+          '2.0.0',
+          '1.2.0',
+          '1.1.0',
+          '1.0.2',
+          '1.0.1',
+          '1.0.0'
+        ]));
+      });
+
+      afterEach(function() {
+        this.getRemoteVersionsStub.restore();
+      });
+
+      it('should be able to resolve the latest v1 version with a semver range', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION, {
+          range: '<2.0.0'
+        }).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('1.2.0');
+          done();
+        }).catch(done);
+      });
+
+      it('should be able to resolve the latest v2 version with a semver range', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION, {
+          range: '>=2.0.0 <3.0.0'
+        }).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('2.1.1');
+          done();
+        }).catch(done);
+      });
+
+      it('should be able to resolve the latest v3 version with a semver range', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION, {
+          range: '>=3.0.0'
+        }).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('3.0.1');
+          done();
+        }).catch(done);
+      });
+
+      it('should resolve the latest version if useUnstableChannel is true', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION, {
+          useUnstableChannel: true
+        }).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('3.0.1');
+          done();
+        }).catch(done);
+      });
+
+      it('should resolve the latest version if useUnstableChannel is false', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION, {
+          considerUnstableReleases: false
```

Comment:
If a user has never installed Etcher before, but then installs Etcher 1.0.0-rc.4 could we default useUnstableChannel to being true? And conversely if a user has never installed Etcher before, but then installs Etcher 1.0.0 could we default useUnstableChannel to being false?